stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1275,12 +1313,72 @@
           reward-set-indexes: (list),
           first-reward-cycle: first-reward-cycle,
           lock-period: lock-period,
-          delegated-to: (some tx-sender) })
+          delegated-to: (some tx-sender),
+          signing-key: next-signing-key })
 
       ;; return the lock-up information, so the node can actually carry out the lock.
       (ok { stacker: stacker,
             unlock-burn-height: new-unlock-ht }))))
 
+;; This function allows a node to register a new 'current-signer-set' during the prepare phase
+(define-private (set-current-signer-set (new-signer-set (list 4000 {signer: (buff 33)})))
```

Comment:
There is currently no test coverage for this.  There should be a unit test in `pox_4_tests.rs` that calls this function with some synthetic data, and reads it back out again.

Reply:
Your tests should hit not only the happy path, but also each of the failure paths (and verify that the right error code is returned).